bitnami/grafana/templates/deployment.yaml

```diff
@@ -0,0 +1,172 @@
+        {{- end }}
+        {{- if .Values.datasources.secretName }}
+        - name: datasources
+          mountPath: /opt/bitnami/grafana/conf/provisioning/datasources
+        {{- end }}
+        {{- if and .Values.ldap.enabled .Values.ldap.configMapName }}
+        - name: ldap
+          mountPath: /opt/bitnami/grafana/conf/ldap.toml
+          subPath: ldap.toml
+        {{- end }}
+        {{- range .Values.extraConfigmaps }}
+        - name: {{ .name }}
+          mountPath: {{ .mountPath }}
+          subPath: {{ .subPath | default "" }}
+          readOnly: {{ .readOnly }}
+        {{- end }}
+        ports:
+          - name: dashboard
+            containerPort: 3000
+            protocol: TCP
+        livenessProbe:
+          tcpSocket:
+            port: dashboard
+{{ toYaml .Values.livenessProbe | indent 10 }}
+        readinessProbe:
+          tcpSocket:
+            port: dashboard
+{{ toYaml .Values.readinessProbe | indent 10 }}
+        resources:
+{{- toYaml .Values.resources | nindent 10 }}
+      volumes:
+      - name: data
+      {{- if .Values.persistence.enabled }}
+        persistentVolumeClaim:
+          claimName: {{ .Values.persistence.existingClaim | default (include "grafana.fullname" .) }}
+      {{- end }}
+      {{- if and .Values.ldap.enabled .Values.ldap.configMapName }}
+      - name: ldap
+        configMap:
+          name: {{ .Values.ldap.configMapName }}
+      {{- end }}
+      {{- if .Values.dashboardsProvider.enabled }}
+      - name: dashboards-provider
+        configMap:
+          {{- if .Values.dashboardsProvider.configMapName }}
+          name: {{ .Values.dashboardsProvider.configMapName }}
+          {{- else }}
+          name: {{ include "grafana.fullname" . }}-provider
+          {{- end }}
+      {{- end }}
+      {{- range .Values.dashboardsConfigMaps }}
+      - name: {{ . }}
+        configMap:
+          name: {{ . }}
+      {{- end }}
+      {{- if .Values.datasources.secretName }}
+      - name: datasources
+        secret:
+          secretName: {{ .Values.datasources.secretName }}
+      {{- end }}
+      {{- if .Values.config.useGrafanaIniFile }}
+      - name: grafana-ini
+        configMap:
+          name: {{ required "A configmap containing the grafana.ini file is needed!!" .Values.config.grafanaIniConfigMap }}
+      {{- end }}
+      {{- if .Values.config.useCustomIniFile }}
+      - name: custom-ini
+        configMap:
+          name: {{ required "A configmap containing the custom.ini file is needed!!" .Values.config.customIniConfigMap }}
+      {{- end }}
+      {{- range .Values.extraConfigmaps }}
+      - name: {{ .name }}
+        configMap:
+          name: {{ .name }}
+      {{- end }}
+{{- with .Values.nodeSelector }}
+      nodeSelector:
+{{- toYaml . | indent 8 }}
+{{- end }}
+{{- with .Values.affinity }}
+      affinity:
+{{- toYaml . | indent 8 }}
+{{- end }}
+{{- with .Values.tolerations }}
+      tolerations:
+{{- toYaml . | indent 8 }}
+{{- end }}
```

Comment:
These parameters should be evaluated as templates (check other charts)

Reply:
From the Wordpress Chart:
```
    {{- if .Values.nodeSelector }}
      nodeSelector:
{{ toYaml .Values.nodeSelector | indent 8 }}
      {{- end -}}
    {{- with .Values.affinity }}
      affinity:
{{ toYaml . | indent 8 }}
    {{- end }}
    {{- with .Values.tolerations }}
      tolerations:
{{ toYaml . | indent 8 }}
{{- end }}
```